src/system-utils.lisp

```diff
@@ -0,0 +1,27 @@
+(in-package :lem-core)
+
+(defun maybe-load-systems (systems &key (silent t) verbose error-on-failure-p)
+  "Load system SYSTEMS, potentially downloading them from an external
+ repository.  SYSTEMS may be a single system or a list of
+ systems. Loader behavior is modified by VERBOSE and SILENT."
+  (unless (listp systems)
+    (setf systems (list systems)))
+  (handler-case
```

Comment:
I think that when you use handler-case, the stack is rolled back and you cannot see the error content of the contents of the load-system.

How about using hanlder-bind?
```common-lisp
(handler-bind ((error (lambda (e)
                        (unless error-on-failure-p
                          (return-from maybe-load-systems nil)))))
  ...)
```